include/libopencm3/stm32/fdcan.h

```diff
@@ -0,0 +1,1420 @@
+#define FDCAN_IE_RF0LE BIT3
+/* Rx FIFO 0 full interrupt enable */
+#define FDCAN_IE_RF0FE BIT2
+/* Rx FIFO 0 watermark reached interrupt enable */
+#define FDCAN_IE_RF0WE BIT1
+/* Rx FIFO 0 new message interrupt enable */
+#define FDCAN_IE_RF0NE BIT0
+
+/* --- FDCAN_ILS: FDCAN interrupt line select register ---------------------- */
+/* Access to reserved address line */
+#define FDCAN_ILS_ARAL BIT29
+/* Protocol error in data phase line */
+#define FDCAN_ILS_PEDL BIT28
+/* Protocol error in arbitration phase line */
+#define FDCAN_ILS_PEAL BIT27
+/* Watchdog interrupt Line */
+#define FDCAN_ILS_WDIL BIT26
+/* Bus_Off status */
+#define FDCAN_ILS_BOL BIT25
+/* Warning status interrupt Line */
+#define FDCAN_ILS_EWL BIT24
+/* Error passive interrupt line */
+#define FDCAN_ILS_EPL BIT23
+/* Error logging overflow interrupt line */
+#define FDCAN_ILS_ELOL BIT22
+/* Message stored to dedicated Rx buffer interrupt line */
+#define FDCAN_ILS_DRXL BIT19
+/* Timeout occurred interrupt Line */
+#define FDCAN_ILS_TOOL BIT18
+/* Message RAM access failure interrupt line */
+#define FDCAN_ILS_MRAFL BIT17
+/* Timestamp wraparound interrupt line */
+#define FDCAN_ILS_TSWL BIT16
+/* Tx event FIFO element Lost interrupt line */
+#define FDCAN_ILS_TEFLL BIT15
+/* Tx event FIFO full interrupt line */
+#define FDCAN_ILS_TEFFL BIT14
+/* Tx event FIFO watermark reached interrupt line */
+#define FDCAN_ILS_TEFWL BIT13
+/* Tx event FIFO new entry interrupt line */
+#define FDCAN_ILS_TEFNL BIT12
+/* Tx FIFO empty interrupt Line */
+#define FDCAN_ILS_TFEL BIT11
+/* Transmission cancellation finished interrupt line */
+#define FDCAN_ILS_TCFL BIT10
+/* Transmission completed interrupt line */
+#define FDCAN_ILS_TCL BIT9
+/* High priority message interrupt line */
+#define FDCAN_ILS_HPML BIT8
+/* Rx FIFO 1 message lost interrupt line */
+#define FDCAN_ILS_RF1LL BIT7
+/* Rx FIFO 1 full interrupt line */
+#define FDCAN_ILS_RF1FL BIT6
+/* Rx FIFO 1 watermark reached interrupt line */
+#define FDCAN_ILS_RF1WL BIT5
+/* Rx FIFO 1 new message interrupt line */
+#define FDCAN_ILS_RF1NL BIT4
+/* Rx FIFO 0 message lost interrupt line */
+#define FDCAN_ILS_RF0LL BIT3
+/* Rx FIFO 0 full interrupt line */
+#define FDCAN_ILS_RF0FL BIT2
+/* Rx FIFO 0 watermark reached interrupt line */
+#define FDCAN_ILS_RF0WL BIT1
+/* Rx FIFO 0 new message interrupt line */
+#define FDCAN_ILS_RF0NL BIT0
+
+/* --- FDCAN_ILE: FDCAN interrupt line enable register ---------------------- */
+/* Enable interrupt Line 1 */
+#define FDCAN_ILE_EINT1 BIT1
+/* Enable interrupt Line 0 */
+#define FDCAN_ILE_EINT0 BIT0
+
+/* --- FDCAN_GFC: FDCAN global filter configuration register ---------------- */
+/* Accept non-matching frames standard */
+enum fdcan_gfc_anfs {
+	FDCAN_GFC_ANFS_ACCEPT_FIFO0	= 0x0,
+	FDCAN_GFC_ANFS_ACCEPT_FIFO1	= 0x1,
+	FDCAN_GFC_ANFS_REJECT		= 0x2,
+	FDCAN_GFC_ANFS_MASK			= 0x3,
+};
+#define FDCAN_GFC_ANFS_SHIFT 4
+/* Accept non-matching frames extended */
+enum fdcan_gfc_anfe {
```

Comment:
is there much value in these two enums being separately named? I'm not a heavy can user, would it be nicer/equally sane from an enduser point of view to just have FDCAN_GFC_ACCEPTFIFO0/1/REJECT/MASK instead of duped for anfs and anfe?

Reply:
I was on the fence about this as well. I'll make them consistent.